### DISCUSSION TITLE
build-bootstrap.sh for android-5

### DISCUSSION THREAD

**Discussion by chenqian-dev** [termux/termux-packages#18388] [Q&A] [answered]
How can I use the build-bootstraps.sh script to compile a package suitable for android-5

**Comment by agnostic-apollo** [comment, marked as answer]
The `android-5` did not have `build-bootstrap.sh`, it was added later. It only has `generate-bootstrap.sh`.

https://github.com/termux/termux-packages/blob/android-5/scripts/generate-bootstraps.sh

You could probably write one yourself for it based on `build-bootstrap.sh` of `master` branch, although it has various issues. Or host a repo and use  existing `generate-bootstrap.sh`.

You will also need [`termux-package-sources-legacy-24.12.2019.tar`](https://archive.org/download/termux-repositories-legacy) as many upstream package source urls in `android-5` branch would be inactive.
